# Goli Restaurant Analysis

There are 3 datasets provided for Analysis. We import all of them as pandas dataframes.

In [1]:
import pandas as pd
import numpy as np
df1 = pd.read_csv(r'C:\Users\Udish Nagpal\Desktop\data science\Market Pricing.csv')
df2 = pd.read_csv(r'C:\Users\Udish Nagpal\Desktop\data science\Customer Bill Detail.csv')
df3 = pd.read_csv(r'C:\Users\Udish Nagpal\Desktop\data science\Customer Order Item Details.csv')

The df1 Dataframe has the 'Market Pricing' details which have the pricing detials for all the dishes and their categories.


The df2 Dataframe has the bill details for a particular order.


The df3 Dataframe has the individual item details corresponding to a particular order including item quantity.

In [3]:
df1 = df1.replace(np.nan, '', regex=True)
df1

Restaurant        Menu Header                            Menu Item Rate  \
0          12 e       Veg Starters                         Paneer Tikka  190   
1          12 e       Veg Starters               Paneer Peshawari Tikka  200   
2          12 e       Veg Starters               Paneer Kalimirch Tikka  190   
3          12 e       Veg Starters                   Paneer Malai Tikka  200   
4          12 e       Veg Starters            Cheese Chilly Onion Kabah  160   
5          12 e       Veg Starters                       Mushroom Tikka  160   
6          12 e       Veg Starters                    Subz Sholai Kabab  160   
7          12 e   Non Veg Starters                           Murg Tikka  240   
8          12 e   Non Veg Starters                     Murg Malai Tikka  250   
9          12 e   Non Veg Starters                    Murg Pahadi Tikka  240   
10         12 e   Non Veg Starters                    Murg Afgani Tikka  250   
11         12 e   Non Veg Starters                Fish Aiurilsari Tikka  260   
12         12 e   Non Veg Starters  Fish Tikka/ Kalimirch/Garlic/ Malai  260   
13         12 e   Non Veg Starters                          Fish Finger  260   
14         12 e   Non Veg Starters                Tandoori Chicken Half  160   
15         12 e   Non Veg Starters                         Bhuna Mutton  260   
16         12 e   Non Veg Starters                        Bhuna Chicken  240   
17         12 e      Indian Breads                  Plain Tandori Roti    15   
18         12 e      Indian Breads                  Butter TandoriRoti    20   
19         12 e      Indian Breads                          Plain Naan    25   
20         12 e      Indian Breads                         Butter Naan    30   
21         12 e      Indian Breads                      Lachha Paiatha    30   
22         12 e      Indian Breads                    Plain Garlic Naan   35   
23         12 e      Indian Breads                  Butter Garlic Naan    40   
24         12 e      Indian Breads                    Cheese Stuff Naan   60   
25         12 e  Main Course - Veg                              Dal fry  120   
26         12 e  Main Course - Veg                          Dal Tad lea  140   
27         12 e  Main Course - Veg                          Dal Makhani  150   
28         12 e  Main Course - Veg                 Butter Paneer Masala  190   
29         12 e  Main Course - Veg                  Paneer Tikka Masala  200   
...         ...                ...                                  ...  ...   
6855       Zaia  Special Corn Food                    Mirch Makai Pulao  130   
6856       Zaia  Special Corn Food              Corn Biryani with Raita  140   
6857       Zaia     Accompaniments      Green Salad and Kachumbar Salad   50   
6858       Zaia     Accompaniments                          Onion Salad   30   
6859       Zaia     Accompaniments                            Aloo Chat   60   
6860       Zaia     Accompaniments                         Peanut Chat    70   
6861       Zaia     Accompaniments                         Papad Masala   25   
6862       Zaia     Accompaniments                        Papad Roasted   20   
6863       Zaia     Accompaniments                            Papad Fry   20   
6864       Zaia     Accompaniments                    Green  Chilly Fry   30   
6865       Zaia     Accompaniments                        Boondi Raita    60   
6866       Zaia     Accompaniments                            Veg Raita   60   
6867       Zaia     Accompaniments                         Pudina Raita   60   
6868       Zaia     Accompaniments                      Pineapple Raita   80   
6869       Zaia     Accompaniments                          Fruit Raita   80   
6870       Zaia     Accompaniments                     Butter [1 Ounce]   30   
6871       Zaia          Beverages                        Panjabi Laasi   50   
6872       Zaia          Beverages                          Cold Coffee   50   
6873 

In [4]:
df1 = df1.replace(np.nan, '', regex=True)
ef  = df1
ef['Menu Item']= df1['Menu Item'].str.lower() 

ef['Menu Item'] = ef['Menu Item'].str.replace(r"\(.*\)","")

In the above operations, we try to cut the variations in dish names which may vary in written form. For Eg: 'CHICKEN MASALA Half' is same as 'Chicken masala (Half)'

EXTRA POINTS OPPORTUNITY(1ST):

Below we notice that price for some dishes of Restaurants are not mentioned.

In [10]:
Missing = df1[df1['Rate'] == '']
Missing

Restaurant           Menu Header           Menu Item Rate  \
197        Alol            VEGETARIAN         malaikofta         
198        Alol            VEGETARIAN              mixveg        
199        Alol            VEGETARIAN           veg kofta        
200        Alol            VEGETARIAN          gobhi aloo        
201        Alol            VEGETARIAN   chanapindi masala        
202        Alol            VEGETARIAN          kaju curry        
203        Alol            VEGETARIAN            kadaiveg        
204        Alol            VEGETARIAN   dum aloo maharani        
205        Alol            VEGETARIAN          corn palak        
206        Alol            VEGETARIAN     mushroom masala        
207        Alol            VEGETARIAN      palak mushroom        
208        Alol            VEGETARIAN         palak kofta        
587        Awen              DESSERTS           ice cream        
1466       Foot              Non Veg.       chicken fish         
1467       Foot              Non Veg.      tandoori fish         
1569       Foot    Sea Food (Seasnal)        fish masala         
1570       Foot    Sea Food (Seasnal)           fish fry         
1571       Foot    Sea Food (Seasnal)           fish fry         
1572       Foot    Sea Food (Seasnal)           fish fry         
5016       Pisa         Cold Bevirage       mineral water        
5017       Pisa         Cold Bevirage               soda         
5018       Pisa         Cold Bevirage          soft drink        
5019       Pisa         Cold Bevirage      soft drink can        
5144       Pisa       CHINESE NON VEG      prawns chilly         
5145       Pisa       CHINESE NON VEG  prawns golden fry         
5203       Pisa        Non Veg. Kabeb     tandoori prawns        
5205       Pisa        Non Veg. Kabeb        pamplate fry        
5206       Pisa        Non Veg. Kabeb   tandoori pamplate        
5293       Pisa  Main Coures Non Veg.         prawnscurry        
5294       Pisa  Main Coures Non Veg.    prawns peshawari        
5400       Pisa                  Beer     haywards 5000          
5402       Pisa                  Beer           fosters          
5410       Pisa                  Beer            miller          
5411       Pisa                  Wine      jacobe creek          
5412       Pisa                  Wine      jacobe creek          
5423       Pisa                  Wine          champagne         
5424       Pisa                  Wine          champagne         

                                  What people love here  
197                                          Fish Tikka  
198                                          Fish Tikka  
199                                          Fish Tikka  
200                                          Fish Tikka  
201                                          Fish Tikka  
202                                          Fish Tikka  
203                                          Fish Tikka  
204                                          Fish Tikka  
205                                          Fish Tikka  
206                                          Fish Tikka  
207                                          Fish Tikka  
208                                          Fish Tikka  
587                             Paneer Tikka, Dal Tadka  
1466                                                     
1467                                                     
1569                                                     
1570                                                     
1571                                                     
1572                                                     
5016  Butter Chicken, Boti Chicken, Tandoori Chicken...  
5017  Butter Chicken, Boti Chicken, Tandoori Chicken...  
5018  Butter Chicken, Boti Chicken, Tandoori Chicken...  
5019  Butter Chicken, Boti Chicken, Tandoori Chicken...  
5144  Butter Chicken, Boti Chicken, Tandoori Chicken...  
5145  Butter Chicken, Boti Chicken, Tandoori C

EXTRA POINTS OPPORTUNITY(2ND)

For some of the orders if we examine the subtotal and delivery cost in the Order Detail Dataset. The condition:
 Total Bill = Subtotal + Delivery Cost isn't valid. 
 There are errors in the TOTAL BILL collected column

In [88]:
wr = []
print(' Calculated and Actual Comparison')
for a in range(df2.shape[0]):
    sum = df2.get_value(a,'Subtotal') + df2.get_value(a,'Delivery Amt')
    if (round(sum) != df2.get_value(a,'Total Bill')):
        print(a , round(sum),'  ',df2.get_value(a,'Total Bill'))
        wr.append(a)
wr

 Calculated and Actual Comparison
119 487.0    488
974 454.0    455
4349 348.0    349
4352 126.0    127
4354 102.0    103
4501 102.0    103
4955 82.0    83
5185 80.0    81
5189 80.0    81
5278 90.0    91
5773 102.0    103
6389 370.0    200
6390 272.0    112
6391 352.0    232
6392 260.0    108
6393 395.0    330
6394 295.0    185
6395 448.0    128
6396 235.0    75
6397 479.0    200
6399 239.0    104
6400 698.0    500
6401 550.0    360
6402 335.0    160
6403 531.0    259
6404 428.0    408
6405 448.0    128
6406 492.0    152
6407 160.0    80
6408 630.0    360
6409 260.0    100
6410 170.0    20
6411 300.0    160
6412 268.0    140
6413 320.0    176
6414 284.0    220
6415 428.0    128
6416 250.0    40
6417 416.0    232
6419 536.0    256
6420 428.0    160
6421 406.0    150
6422 528.0    256
6423 408.0    72
6424 288.0    160
6425 455.0    160
6426 310.0    160
6427 394.0    144
6428 384.0    160
6429 486.0    320
6430 250.0    150
6431 376.0    136
6432 432.0    160
6433 708.0    408
6434 488.

D:\Technical\lib\site-packages\ipykernel_launcher.py:4: FutureWarning: get_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead
  after removing the cwd from sys.path.
D:\Technical\lib\site-packages\ipykernel_launcher.py:5: FutureWarning: get_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead
  """
D:\Technical\lib\site-packages\ipykernel_launcher.py:6: FutureWarning: get_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead
  


7314 104.0    105
7382 84.0    85
7581 84.0    85
7599 79.0    80
7611 74.0    75
7764 84.0    85
7772 94.0    95
7788 84.0    85
7798 84.0    85
7802 58.0    59
7811 64.0    65
7829 94.0    95
7839 76.0    77
7846 88.0    89
7892 84.0    85
7914 90.0    91
7948 84.0    85
7972 54.0    55
7974 58.0    59
7985 84.0    85
8007 54.0    55
8052 76.0    77
8091 94.0    95
8092 76.0    77
8102 58.0    59
8136 64.0    65
8156 64.0    65
8234 114.0    115
8239 94.0    95
8327 94.0    95


[119,
 974,
 4349,
 4352,
 4354,
 4501,
 4955,
 5185,
 5189,
 5278,
 5773,
 6389,
 6390,
 6391,
 6392,
 6393,
 6394,
 6395,
 6396,
 6397,
 6399,
 6400,
 6401,
 6402,
 6403,
 6404,
 6405,
 6406,
 6407,
 6408,
 6409,
 6410,
 6411,
 6412,
 6413,
 6414,
 6415,
 6416,
 6417,
 6419,
 6420,
 6421,
 6422,
 6423,
 6424,
 6425,
 6426,
 6427,
 6428,
 6429,
 6430,
 6431,
 6432,
 6433,
 6434,
 6435,
 6436,
 6437,
 6438,
 6439,
 6440,
 6442,
 6443,
 6444,
 6445,
 6446,
 6447,
 6448,
 6449,
 6450,
 6451,
 6452,
 6453,
 6454,
 6455,
 6456,
 6457,
 6459,
 6460,
 6461,
 6462,
 6463,
 6464,
 6465,
 6467,
 6468,
 6469,
 6470,
 6471,
 6472,
 6473,
 6474,
 6475,
 6477,
 6478,
 6479,
 6480,
 6481,
 6482,
 6483,
 6484,
 6485,
 6486,
 6487,
 6488,
 6489,
 6490,
 6491,
 6492,
 6493,
 6494,
 6495,
 6496,
 6497,
 6498,
 6499,
 6500,
 6501,
 6502,
 6503,
 6504,
 6505,
 6506,
 6507,
 6508,
 6509,
 6510,
 6511,
 6512,
 6513,
 6514,
 6515,
 6516,
 6517,
 6518,
 6519,
 6520,
 6521,
 6522,
 6523,
 6524,
 6525,
 6526,
 

First few miscomparisons are permissible but the miscomparisons from 6389 to 6757 are huge.
There's some mistake in the data.

Now we want to analyse the market condition for GOLI's Restaurant and help him in keeping his products at a competitive price.

In [18]:
Golidf  = df1[df1['Restaurant'] == 'Goli']
Golidf

Restaurant             Menu Header  \
1904       Goli             Main Course   
1905       Goli             Main Course   
1906       Goli             Main Course   
1907       Goli             Main Course   
1908       Goli     Tandoori and Kebabs   
1909       Goli     Tandoori and Kebabs   
1910       Goli        Rice and Biryani   
1911       Goli        Rice and Biryani   
1912       Goli        Rice and Biryani   
1913       Goli        Rice and Biryani   
1914       Goli             Rice Combos   
1915       Goli             Rice Combos   
1916       Goli             Rice Combos   
1917       Goli                Platters   
1918       Goli                Platters   
1919       Goli                Platters   
1920       Goli               Breakfast   
1921       Goli               Breakfast   
1922       Goli               Breakfast   
1923       Goli               Breakfast   
1924       Goli                Starters   
1925       Goli                Starters   
1926       Goli                Starters   
1927       Goli                Starters   
1928       Goli                Starters   
1929       Goli                Starters   
1930       Goli                Starters   
1931       Goli                Starters   
1932       Goli                Starters   
1933       Goli                Starters   
...         ...                     ...   
2055       Goli                    Rice   
2056       Goli                    Rice   
2057       Goli                 Biryani   
2058       Goli                 Biryani   
2059       Goli                 Biryani   
2060       Goli                 Biryani   
2061       Goli                 Biryani   
2062       Goli                 Biryani   
2063       Goli                 Biryani   
2064       Goli                   Rolls   
2065       Goli                   Rolls   
2066       Goli                   Rolls   
2067       Goli                   Rolls   
2068       Goli          Chinese Dishes   
2069       Goli          Chinese Dishes   
2070       Goli          Chinese Dishes   
2071       Goli          Chinese Dishes   
2072       Goli          Chinese Dishes   
2073       Goli          Chinese Dishes   
2074       Goli          Chinese Dishes   
2075       Goli  Fried Rice and Noodles   
2076       Goli  Fried Rice and Noodles   
2077       Goli  Fried Rice and Noodles   
2078       Goli          Accompaniments   
2079       Goli          Accompaniments   
2080       Goli          Accompaniments   
2081       Goli          Accompaniments   
2082       Goli          Accompaniments   
2083       Goli          Accompaniments   
2084       Goli          Accompaniments   

                                      Menu Item Rate  \
1904                         chicken curry full  220   
1905                         chicken curry half  120   
1906                        butter chicken half  180   
1907                        butter chicken full  340   
1908                      tandoori chicken half  160   
1909                      tandoori chicken full  290   
1910  hyderabadi chicken fry piece biryani half  160   
1911  hyderabadi chicken fry piece biryani full  300   
1912         hyderabadi mutton dum biryani full  300   
1913         hyderabadi mutton dum biryani half  160   
1914                           rajma rice combo  100   
1915                           chole rice combo  100   
1916                    lemon rice combo [full]   80   
1917                               veg platter   500   
1918                      non veg royal platter  700   
1919                            chicken platter  600   
1920                                    omlatte   40   
1921                              bread omlatte   50   
1922                              chole bathure   80   
1923                     chole kulche amritsari   80   
1924                  chicken garlic tikka half  120   
1925                  chicken garlic tikka full  190   
1926                  chicken pudina tikka full  1

In [13]:
Goli_dishes = Golidf['Menu Item'].unique().tolist()

Now for a competitive pricing of the Dishes, we analyse and compare the prices at various other restaurants. We will compare our prices with the AVERAGE MARKET PRICE for that particular dish. We also compare it with the MEDIAN PRICE and the MODE PRICE. It should be always be close to these central identities specially the MEAN and MEDIAN. Also, if there isn't a unique most frequently occuring value, we take the lowest MODE into consideration.

NOTE: If our price for a certain dish is higher than all the three MEAN, MEDIAN AND MODE and that particular dish is not the most sold or loved dish of our Restaurant, then we may need to consider lowering the prices. The same goes if our dish is at too low cost.

The Prices are displayed in the order.
GOLI'S PRICE, MEAN, MEDIAN, MODE

In [26]:
import statistics
d_pricing = {}
for a in Goli_dishes:
    Goli_ind = []
    val  = 0
    ind = []
    l = []
    Goli_ind = Golidf.index[Golidf['Menu Item'] == a].tolist()
    Goli_Price = Golidf.get_value(Goli_ind[0],'Rate')
    ind  = df1.index[df1['Menu Item'] == a].tolist()
    for t in ind:
        if((df1.get_value(t,'Rate') == '')) :
           
            ind.remove(t)
        else:
            
            try:
                if(int(df1.get_value(t,'Rate'))<1):
                    ind.remove(t)
                
                else:
                    try:
                        
                        l.append(int(ef.get_value(t,'Rate')))
                
                        val = val + int(ef.get_value(t,'Rate'))
                    except ValueError:
                        ind.remove(t)
            except ValueError:
                ind.remove(t)
    median = statistics.median(l)
    mean = statistics.mean(l)
    try:
        mode = statistics.mode(l)
    except:
        mode = min(l)
    d_pricing[a] = [Goli_Price,mean,median,mode]
print('Goli,s Price,Mean,Median,Mode')
d_pricing
        

D:\Technical\lib\site-packages\ipykernel_launcher.py:9: FutureWarning: get_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead
  if __name__ == '__main__':
D:\Technical\lib\site-packages\ipykernel_launcher.py:12: FutureWarning: get_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead
  if sys.path[0] == '':
D:\Technical\lib\site-packages\ipykernel_launcher.py:18: FutureWarning: get_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead
D:\Technical\lib\site-packages\ipykernel_launcher.py:24: FutureWarning: get_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead
D:\Technical\lib\site-packages\ipykernel_launcher.py:26: FutureWarning: get_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead


Goli,s Price,Mean,Median,Mode


{'aloo gobhi': ['100', 117.5, 115.0, 100],
 'aloo matar': ['100', 100, 100.0, 100],
 'aloo palak': ['100', 105, 100, 80],
 'aloo paratha': ['40', 37.5, 40.0, 40],
 'bhindi masala': ['120', 141, 145.0, 90],
 'bhuna chicken full': ['280', 280, 280, 280],
 'bhuna chicken half': ['150', 150, 150, 150],
 'bhuna mutton half': ['150', 215, 215.0, 150],
 'bread omlatte': ['50', 50, 50, 50],
 'butter chicken full': ['340', 363.3333333333333, 340.0, 340],
 'butter chicken half': ['180', 200, 190.0, 180],
 'butter naan ': ['25', 30, 27.5, 25],
 'butter tandoori roti ': ['15', 15, 15, 15],
 'cheese garlic paratha': ['40', 40, 40, 40],
 'chicken afgani tikka half': ['120', 120, 120, 120],
 'chicken afghani tikka full': ['190', 190, 190, 190],
 'chicken amritsari tikka full': ['190', 190, 190, 190],
 'chicken amritsari tikka half': ['120', 120, 120, 120],
 'chicken chettinad full': ['260', 260, 260, 260],
 'chicken chettinad half': ['140', 140, 140, 140],
 'chicken curry full': ['220', 270, 260.0, 2

For competitive pricing we must carefully observe the prices of the dishes that are loved at GOLI's Restaurant.

2. HOW TO INCREASE THE REVENUE? Strategies?

In [37]:
df2['Type'].value_counts()

Home Delivery    6035
Dine In          1395
Take Away         948
Name: Type, dtype: int64

Firstly, we need to Increasing the Table Turnover Rate, i.e, We need to increase the DINE_IN customers. We have a lack of Dine in customers as compared to Home Delivery. However, this does not mean that you should rush customers through their experience.

Secondly, we need to promote our most frequent customers, specially the ones who DINE_IN.

In [45]:
customer = df2['Customer ID'].unique().tolist()
customer.remove('UnKnown')
customer

['38271357206',
 '249028087096',
 '249225356278',
 '249229152321',
 '231587765414',
 '249229190637',
 '69425463848',
 '249229174207',
 '248949293714',
 '297461909752',
 '245837713261',
 '307549806093',
 '35326927730',
 '247993690628',
 '306686823939',
 '248949292164',
 '69425465491',
 '300449762819',
 '256355146873',
 '249229207563',
 '279309402471',
 '245837713199',
 '69427931448',
 '69419339395',
 '249028101790',
 '249028100674',
 '69427966106',
 '304606418279',
 '248943910316',
 '249224643898',
 '69428028602',
 '246127156169',
 '248949288382',
 '69427933649',
 '69438915058',
 '69427866224',
 '249025404697',
 '249229172750',
 '35326922832',
 '249229185119',
 '69427866751',
 '246127156386',
 '249225350357',
 '248949286305',
 '248943904240',
 '249229166364',
 '249229169991',
 '248943908611',
 '248949302022',
 '304614580827',
 '103512029998',
 '249229172781',
 '69419591332',
 '249225351256',
 '69428031144',
 '283082170113',
 '248949283980',
 '249028086166',
 '194108736708',
 '3046186234

In [83]:
df2['Customer ID'].value_counts()

    

UnKnown         1759
38271357206      689
34478675877      174
38271354044      108
289200068721      96
34099752616       64
34819954629       47
34816569057       42
258819372616      26
251380891616      17
302408281315      16
304606418279       9
304732564657       6
249025417190       6
224005569437       5
231587765414       5
291596912833       5
34785231188        5
288319320281       4
247974745009       4
248949286801       4
301115486889       4
69427921776        4
246117310135       4
303822923504       4
245850959499       3
249229160567       3
69715692668        3
69428032012        3
69427741387        3
                ... 
249450686009       1
249225349892       1
248943905232       1
250206181770       1
69715728380        1
273328793284       1
250228999382       1
248949283608       1
249229173060       1
249229204432       1
297111743703       1
69419528743        1
245850958631       1
250206172966       1
125401153961       1
250230418314       1
138630663741 

38271357206 :689
34478675877 :174
38271354044 :108
289200068721 :96
34099752616  :64
34819954629  :47
34816569057  :42

Customers with these customer ID's should be given Loyalty Offers so that they particpate even more and DINE_IN regularly. 
These customers are frequent visitors to our restaurant.

STRATEGY 3:
We may introduce some 'Happy Hours' so that for some time duration daily, we may offer some attractive discounts that may attract some customers.
Eg; In daytime, when there's less crowd on weekdays, we may offer some good discounts on some of our loved dishes like
'Hyderabadi Chicken Fry Piece Biryani',
Butter Chicken',
'Hyderabadi Mutton Dum Biryani' etc

We may also include the dishes that are to be promoted.

STRATEGY 4: 
We may catch in on Festival Dates and promote the festive items in our menu associated with that particular festival.
Eg- On Eid, Chicken and Mutton Dishes that are best sold should be on discount.
Similarly South Indian dishes should be on discount on Pongal.

In [47]:
dishcount = df3['Dish Name'].unique()

In [49]:
dishcount.tolist()

['Bhel',
 'Cheese Chutney Half',
 'Cold Drink',
 'Cold Drink s',
 'Hot Coffee',
 'Masala Cheese Maggi',
 'Tomato Soup',
 'water bottle',
 'Chicken Home Style',
 'Chicken Tikka',
 'Tandoori Roti',
 'Fish Curry',
 'Hyderabadi Mutton Dum Biryani',
 'Plain Rice',
 'Chicken Handi',
 'Sev Tamater',
 'Tandoori Roti Butter',
 'Dhaniya Chicken',
 'Hyderabadi Chicken Fry Piece Biryani',
 'Chicken Manchurian',
 'Chicken Amritsari Tikka',
 'Chicken masala',
 'Fried Rice',
 'Kadhai Chicken',
 'Tawa Roti',
 'Chicken Tikka Masala',
 'Butter Chicken',
 'Fish Fry',
 'Chicken Malai Tikka',
 'Butter nan',
 'Egg Curry',
 'muttor Mushroom',
 'Veg Biryani',
 'Butter Tawa Roti',
 'Mutton Home Style',
 'Jeera Rice',
 'Mutton Curry',
 'Butter maggi',
 'Mutton Bati',
 'Tandoori Chicken',
 'Aloo Gobhi',
 'Dal tadhka',
 'Chicken Curry',
 'Lachchha Paratha',
 'Mutton Bhuna',
 'Chicken Nilgir',
 'Kadhai Paneer',
 'Tandoori Chicken Afgani',
 'Chicken Garlic Tikka',
 'Chicken Chettinad',
 'Garlic Nan',
 'Keema Masala

In [50]:
love = {}
for a in dishcount:
    sum =0
    w = df3.index[df3['Dish Name'] == a].tolist()
    for q in w:
        sum  = sum+ df3.get_value(q,'Quantity')
        love[a] = sum
love

D:\Technical\lib\site-packages\ipykernel_launcher.py:6: FutureWarning: get_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead
  


{'Aloo Gobhi': 29,
 'Aloo Muttor': 36,
 'Aloo Palak': 8,
 'Aloo Paratha': 208,
 'Bati': 13,
 'Benjo': 37,
 'Bhel': 41,
 'Bhindi masala': 59,
 'Bhuna Chicken': 132,
 'Boiled Egg': 29,
 'Boiled Fry': 4,
 'Bread Butter Toast': 2,
 'Bread Omlette': 22,
 'Bread butter cheese': 5,
 'Butter Cheese maggi': 5,
 'Butter Chicken': 823,
 'Butter Khichdhi': 37,
 'Butter Milk': 55,
 'Butter Tawa Roti': 825,
 'Butter maggi': 12,
 'Butter nan': 1040,
 'Cheese Chutney Half': 21,
 'Cheese Garlic Paratha': 59,
 'Cheese maggi': 19,
 'Chicken Afghani Tikka': 82,
 'Chicken Amritsari Tikka': 80,
 'Chicken Chettinad': 43,
 'Chicken Clear Soup': 4,
 'Chicken Curry': 1013,
 'Chicken Curry [2 piece] with 3 Whole Wheat Tandoori Roti [Serves 1]': 4,
 'Chicken Fried Rice': 68,
 'Chicken Fry': 56,
 'Chicken Garlic Tikka': 185,
 'Chicken Handi': 237,
 'Chicken Home Style': 276,
 'Chicken Home Style [2 piece] with 3 Whole Wheat Tandoori Roti [Serves 1]': 6,
 'Chicken Hot n Sour': 2,
 'Chicken Kolhapuri': 180,
 'Chicke

In [51]:
import operator
sorted_x = sorted(love.items(), key=operator.itemgetter(1))
sorted_x

[('Veg Hot n Sour', 1),
 ('Veg Roll & Biryani Combo', 1),
 ('Vegetable cheese', 1),
 ('Vegetable Cheese Half', 1),
 ('Green Tea', 1),
 ('Egg Boiled Fry', 1),
 ('Singapuri Noodles', 1),
 ('Veg Lemon Coriander', 1),
 ('Masala Mayo Cheese Half', 1),
 ('Mushroom Burger', 1),
 ('Paneer Bhurji', 1),
 ('Chicken Rara [8 Piece]', 1),
 ('Veg Chilly Garlic Noodles', 1),
 ('Bread Butter Toast', 2),
 ('Vegetable Mayo cheese', 2),
 ('Chicken Hot n Sour', 2),
 ('chicken masala full', 2),
 ('egg masala', 2),
 ('Non Veg Royal Platter', 2),
 ('Peanut chaat', 2),
 ('Keema Kaleji', 2),
 ('Sev Tamatar with 3 Whole Wheat Tandoori Roti [Serves 1]', 2),
 ('Paneer Lababdaar', 2),
 ('Chicken Monchow', 3),
 ('Chicken Platter', 3),
 ('Veg Monchow', 3),
 ('Veg Soup', 3),
 ('Shimla cheese Paneer Half', 3),
 ('Paneer Tikka Full', 3),
 ('Mutton Home Style [2 piece] with 3 Whole Wheat Tandoori Roti [Serves 1]',
  3),
 ('Masala Cheese', 4),
 ('Kaju chicken', 4),
 ('Choclate Shake', 4),
 ('Boiled Fry', 4),
 ('Veg Platte

'Hyderabadi Chicken Fry Piece Biryani','Butter nan','Butter Chicken','Hyderabadi Mutton Dum Biryani','Tandoori Chicken','Kadhai Chicken','Veg Biryani','Chicken Home Style','Chole Kulche Amritsari'.
These are some of the most loved dishes.

'Egg Biryani','Dal tadhka','Paneer Butter Masala', 'Mutton Home Style','Chilly Paneer' etc which have medium frequency should be promoted more.


Dishes like 'Veg Roll & Biryani Combo' 'Vegetable cheese' 'Egg Boiled Fry' 'Singapuri Noodles' 'Mushroom Burger'
'Paneer Bhurji' 'Chicken Rara [8 Piece]' 'Chicken Hot n Sour' need to be removed. These dishes are like liabilities. They aren't getting any sort of customers'

Combos can be made out of dishes that , one of which is being Sold Heavily and other which should be promoted.
Eg - 
 Garlic Nan(314) with Butter Chicken(823)
 Cold Coffee with 'Hyderabadi Chicken Fry Piece Biryani'
 Jeera Rice with Chicken Curry
 Mutton Bhuna with 'Hyderabadi Mutton Dum Biryani'

CUSTOMER SEGMENTS

By What They Order

In [64]:
eval = dict.fromkeys(['Chicken based','Veg','Mutton based','Noodle based','Beverages'],0)
for a in love:
    
    if('Chicken' in a ):
        eval['Chicken based']+=love[a]
    elif('Mutton' in a):
        eval['Mutton based']+= love[a]
    elif('Noodles' in a or 'maggi' in a):
        eval['Noodle based'] += love[a]
    if('Veg' in a or 'Paneer' in a or 'Dal' in a or 'Mushroom' in a or 'Vegetable' in a or 'Corn' in a or 'Chole' in a):
        eval['Veg'] += love[a]
    if('Tea' in a or 'Coffee' in a or 'Drink' in a or 'water' in a):
        eval['Beverages'] +=love[a]
eval
        

{'Beverages': 1393,
 'Chicken based': 6933,
 'Mutton based': 1474,
 'Noodle based': 93,
 'Veg': 2203}

By How Much They Spend

We will perform this process by calculating the AVERAGE BILL PER ORDER of the customer. We also calculate the MEDIAN and MODES so that we can define the 3 categories High Spending, Low Spending and Medium Spending

In [65]:
customer

['38271357206',
 '249028087096',
 '249225356278',
 '249229152321',
 '231587765414',
 '249229190637',
 '69425463848',
 '249229174207',
 '248949293714',
 '297461909752',
 '245837713261',
 '307549806093',
 '35326927730',
 '247993690628',
 '306686823939',
 '248949292164',
 '69425465491',
 '300449762819',
 '256355146873',
 '249229207563',
 '279309402471',
 '245837713199',
 '69427931448',
 '69419339395',
 '249028101790',
 '249028100674',
 '69427966106',
 '304606418279',
 '248943910316',
 '249224643898',
 '69428028602',
 '246127156169',
 '248949288382',
 '69427933649',
 '69438915058',
 '69427866224',
 '249025404697',
 '249229172750',
 '35326922832',
 '249229185119',
 '69427866751',
 '246127156386',
 '249225350357',
 '248949286305',
 '248943904240',
 '249229166364',
 '249229169991',
 '248943908611',
 '248949302022',
 '304614580827',
 '103512029998',
 '249229172781',
 '69419591332',
 '249225351256',
 '69428031144',
 '283082170113',
 '248949283980',
 '249028086166',
 '194108736708',
 '3046186234

In [70]:
orderdic = {}
wor = []
owfinal = {}
categor = {}
for t in customer:
    total = 0
    wor = df2.index[df2['Customer ID'] == t].tolist()
    ow = df2[df2['Customer ID'] == t][['Customer ID','Total Bill']]
    for a in wor:
        total = total + df2.get_value(a,'Total Bill')
    owfinal[t] = total/len(wor)
   
        
sd = list(owfinal.values())
print(owfinal)
import statistics
med = statistics.median(sd)
mean = statistics.mean(sd)
mod = statistics.mode(sd)
print('MEDIAN,MODE AND MEAN')
print(med,mod,mean)

D:\Technical\lib\site-packages\ipykernel_launcher.py:10: FutureWarning: get_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead
  # Remove the CWD from sys.path while we load stuff.


{'38271357206': 224.35703918722785, '249028087096': 150.0, '249225356278': 160.0, '249229152321': 160.0, '231587765414': 248.0, '249229190637': 240.0, '69425463848': 155.0, '249229174207': 185.0, '248949293714': 335.0, '297461909752': 185.0, '245837713261': 300.0, '307549806093': 160.0, '35326927730': 190.0, '247993690628': 150.0, '306686823939': 310.0, '248949292164': 175.0, '69425465491': 160.0, '300449762819': 300.0, '256355146873': 160.0, '249229207563': 160.0, '279309402471': 190.0, '245837713199': 350.0, '69427931448': 695.0, '69419339395': 204.0, '249028101790': 160.0, '249028100674': 160.0, '69427966106': 432.0, '304606418279': 163.77777777777777, '248943910316': 360.0, '249224643898': 160.0, '69428028602': 438.0, '246127156169': 339.0, '248949288382': 160.0, '69427933649': 225.0, '69438915058': 230.0, '69427866224': 610.0, '249025404697': 178.0, '249229172750': 150.0, '35326922832': 323.0, '249229185119': 180.0, '69427866751': 160.0, '246127156386': 195.0, '249225350357': 230.

MEDIAN = 200.0 
MODE = 160
MEAN = 237.745



In [74]:
for t in owfinal:
    
    if(owfinal[t] > 400):
        categor[t] = 'High Spending'
    elif(owfinal[t] >200):
        categor[t] = 'Medium Spending'
    else:
        categor[t] ='Low Spending'

In [75]:
categor

{'35326928567': 'Low Spending',
 '38271357206': 'Medium Spending',
 '249028087096': 'Low Spending',
 '249225356278': 'Low Spending',
 '249229152321': 'Low Spending',
 '231587765414': 'Medium Spending',
 '249229190637': 'Medium Spending',
 '69425463848': 'Low Spending',
 '249229174207': 'Low Spending',
 '248949293714': 'Medium Spending',
 '297461909752': 'Low Spending',
 '245837713261': 'Medium Spending',
 '307549806093': 'Low Spending',
 '35326927730': 'Low Spending',
 '247993690628': 'Low Spending',
 '306686823939': 'Medium Spending',
 '248949292164': 'Low Spending',
 '69425465491': 'Low Spending',
 '300449762819': 'Medium Spending',
 '256355146873': 'Low Spending',
 '249229207563': 'Low Spending',
 '279309402471': 'Low Spending',
 '245837713199': 'Medium Spending',
 '69427931448': 'High Spending',
 '69419339395': 'Medium Spending',
 '249028101790': 'Low Spending',
 '249028100674': 'Low Spending',
 '69427966106': 'High Spending',
 '304606418279': 'Low Spending',
 '248943910316': 'Medi

In [89]:
import matplotlib.pylab as plt

lists = sorted(categor.items()) # sorted by key, return a list of tuples

x, y = zip(*lists) # unpack a list of pairs into two tuples

plt.plot(x, y)
plt.show()

To Find the most loved dishes in the market

We analyse 'What people Love here' column of the Market pricing Dataset.

In [76]:
restroname = df1['Restaurant'].unique()

In [77]:
restro_name=restroname.tolist()

In [78]:
dic = {}
for a in restro_name:
    l = df1.index[df1['Restaurant'] == a].tolist()
    dic[a] = df1.get_value(l[0],'What people love here')
    
import math
import numpy as np
lov ={}

for a in dic:
    
    floats = [x for x in dic[a].split(', ')]
    dic[a] = floats
    if dic[a]!='':
        
        for b in dic[a]:
            if b in lov:
                lov[b]+=1
            else:
                lov[b]=1
lov

D:\Technical\lib\site-packages\ipykernel_launcher.py:4: FutureWarning: get_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead
  after removing the cwd from sys.path.


{'': 12,
 'Afghan Chicken': 1,
 'Barbeque Chicken': 1,
 'Biryani': 2,
 'Boti Chicken': 1,
 'Butter Chicken': 11,
 'Butter Naan': 1,
 'Chicken Biryani': 6,
 'Chicken Dum Biryani': 1,
 'Chicken Grill': 1,
 'Chicken Lollipop': 1,
 'Chicken Malai Tikka': 1,
 'Chicken Roll': 1,
 'Chicken Shawarama': 1,
 'Chicken Sukha': 1,
 'Chicken Taka Tak': 1,
 'Chicken Tariwala': 1,
 'Chicken Tenders': 1,
 'Chicken Tikka': 5,
 'Chicken Tikka Biryani': 1,
 'Chicken Tikka Masala': 1,
 'Crispy Corn': 1,
 'Dahi Kebab': 1,
 'Dal Makhani': 4,
 'Dal Tadka': 1,
 'Dum Biryani': 2,
 'Egg Curry': 1,
 'Fish': 2,
 'Fish Tikka': 2,
 'Gulab Jamun': 2,
 'Haleem': 1,
 'Hyderabadi Biryani': 1,
 'Hyderabadi Chicken': 1,
 'Jumbo Prawns': 1,
 'Keema Kaleji': 1,
 'Kiwi Mojito': 1,
 'Lassi': 1,
 'Malai Kofta': 1,
 'Manchow Soup': 1,
 'Mint Mojito': 1,
 'Mutton Bhuna': 1,
 'Mutton Biryani': 2,
 'Mutton Curry': 1,
 'Mutton Rara': 1,
 'Naan': 4,
 'Nalli Nihari': 1,
 'Paan Ice Cream': 1,
 'Paneer Makhani': 1,
 'Paneer Tikka': 1,


In [79]:
sorted_x = sorted(lov.items(), key=operator.itemgetter(1))

In [81]:
sorted_x[:-1]    #since last one is empty string''

[('Paneer Tikka', 1),
 ('Dal Tadka', 1),
 ('Chicken Taka Tak', 1),
 ('Barbeque Chicken', 1),
 ('Chicken Grill', 1),
 ('Chicken Tenders', 1),
 ('Chicken Shawarama', 1),
 ('Hyderabadi Chicken', 1),
 ('Nalli Nihari', 1),
 ('Hyderabadi Biryani', 1),
 ('Chicken Dum Biryani', 1),
 ('Raita', 1),
 ('Chicken Tikka Masala', 1),
 ('Jumbo Prawns', 1),
 ('Paan Ice Cream', 1),
 ('Salads', 1),
 ('Crispy Corn', 1),
 ('Keema Kaleji', 1),
 ('Mutton Bhuna', 1),
 ('Mutton Curry', 1),
 ('Vegetable Biryani', 1),
 ('Kiwi Mojito', 1),
 ('Mint Mojito', 1),
 ('Seekh Kebab', 1),
 ('Egg Curry', 1),
 ('Roti', 1),
 ('Chicken Sukha', 1),
 ('Dahi Kebab', 1),
 ('Manchow Soup', 1),
 ('Chicken Tariwala', 1),
 ('Butter Naan', 1),
 ('Haleem', 1),
 ('Chicken Roll', 1),
 ('Chicken Tikka Biryani', 1),
 ('Salad', 1),
 ('Rolls', 1),
 ('Paneer Makhani', 1),
 ('Malai Kofta', 1),
 ('Lassi', 1),
 ('Boti Chicken', 1),
 ('Chicken Lollipop', 1),
 ('Afghan Chicken', 1),
 ('Chicken Malai Tikka', 1),
 ('Mutton Rara', 1),
 ('Fish Tikka',

Butter Chicken, Tandoori Chicken, Chicken Biryani, Chicken Tikka, are the most loved dishes with respect to the market. So GOLI needs to pay special consideration to analysing these dishes.